### ISSUE TITLE
[BUG] Model metrics plots truncate/miss steps

### ISSUE THREAD

**Issue by Tianmaru** [mlflow/mlflow#11481]
### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

2.11.1

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Linux Ubuntu 22.04.4 LTS
- **Python version**: 3.12.2

### Describe the problem

The plots shown in the "Model Metrics" tab sometimes don't show all logged time steps, even though the metrics have been logged continuously:

For example, the screenshot shows, that TEST_ACC and TEST_AUROC are only plotted from step 2750 to 3000 - all other steps are unavailable and they are also not displayed when zooming out. Even more weird, for TRAIN_ACC and TRAIN_AUROC, only a single value is shown, probably the one from the last step:

![grafik](https://github.com/mlflow/mlflow/assets/27966972/a6ee6c13-c780-4926-8ed9-6ad3de08e18e)
 
However, when I open the full plots for each metric via the overview tab, it shows that the metrics have been actually logged as intended:

![grafik](https://github.com/mlflow/mlflow/assets/27966972/3a5508f9-1ad2-4c74-b48f-5f60ff321005)

### Steps to reproduce the bug

I am not able to describe reproduction steps, since I am not sure what's causing this behavior. I just created a PyTorch model and logged loss, accuracy and auroc via `mlflow.log_metric`.

**Comment by Tianmaru** [comment]
Might be related to #11444

**Comment by daniellok-db** [comment]
I this is related to #11444. Are you logging your metrics at every single step, or only at some intervals?

**Comment by Tianmaru** [comment]
No, they are indeed not logged every single step, since the metrics are computed over one epoch. Meanwhile, I also log the train and test loss, which is logged every step. This might actually be related to the cause, since the same problem is not observed for the loss charts.

**Comment by Tianmaru** [comment]
Since this seems to be a duplicate of an already addressed issue, I will close this issue.
